lib/src/protocol/kawa_h1/mod.rs

```diff
@@ -0,0 +1,1933 @@
+            self.container_frontend_timeout.triggered();
+            return match self.timeout_status() {
+                TimeoutStatus::Request => {
+                    self.set_answer(DefaultAnswerStatus::Answer408, None);
+                    self.writable(metrics)
+                }
+                TimeoutStatus::WaitingForResponse => {
+                    self.set_answer(DefaultAnswerStatus::Answer504, None);
+                    self.writable(metrics)
+                }
+                TimeoutStatus::Response => StateResult::CloseSession,
+                TimeoutStatus::WaitingForNewRequest => StateResult::CloseSession,
+            };
+        }
+
+        if self.backend_token == Some(token) {
+            //info!("backend timeout triggered for token {:?}", token);
+            self.container_backend_timeout.triggered();
+            return match self.timeout_status() {
+                TimeoutStatus::Request => {
+                    error!(
+                        "got backend timeout while waiting for a request, this should not happen"
+                    );
+                    self.set_answer(DefaultAnswerStatus::Answer504, None);
+                    self.writable(metrics)
+                }
+                TimeoutStatus::WaitingForResponse => {
+                    self.set_answer(DefaultAnswerStatus::Answer504, None);
+                    self.writable(metrics)
+                }
+                TimeoutStatus::Response => {
+                    error!(
+                        "backend {:?} timeout while receiving response (cluster {:?})",
+                        self.backend_id, self.cluster_id
+                    );
+                    StateResult::CloseSession
+                }
+                TimeoutStatus::WaitingForNewRequest => StateResult::Continue,
+            };
+        }
+
+        error!("got timeout for an invalid token");
+        StateResult::CloseSession
+    }
+
+    fn cancel_timeouts(&mut self) {
+        self.container_backend_timeout.cancel();
+        self.container_frontend_timeout.cancel();
+    }
+
+    fn print_state(&self, context: &str) {
+        error!(
+            "\
+{} Session(Pipe)
+\tFrontend:
+\t\ttoken: {:?}\treadiness: {:?}\tstate: {:?}
+\tBackend:
+\t\ttoken: {:?}\treadiness: {:?}\tstate: {:?}",
+            context,
+            self.frontend_token,
+            self.frontend_readiness,
+            self.request_stream.parsing_phase,
+            self.backend_token,
+            self.backend_readiness,
+            self.response_stream.parsing_phase
+        );
+    }
+
+    fn shutting_down(&mut self) -> SessionIsToBeClosed {
+        if self.request_stream.is_initial()
+            && self.request_stream.storage.is_empty()
+            && self.response_stream.storage.is_empty()
+        {
+            true
+        } else {
+            self.context.closing = true;
+            false
+        }
+    }
+}
+
+/// Save the backend http response status code metric
```

Comment:
These are 6 words in a row, please rewrite to a more readable and grammaticaly explicit wording, for instance "save the HTTP status code of the backend response as a metric" or a similar verbose way